### PR TITLE
Add PCH for tests

### PR DESCRIPTION
**Pull request by ras0219-msft**
We have a substantial number of TUs in our tests, so they stand to benefit from a PCH.

Unfortunately, PCHs cannot be shared across compilations with different macro settings.

Edit: I've also broken up our `=== Files ===` section into per-target sections, since in the progress of creating this PR I was confused about the number of files that went into each binary.

### CODE REVIEW

**Comment by autoantwort** [comment]
In the future we could use https://cmake.org/cmake/help/latest/command/target_precompile_headers.html (requires cmake 3.16, we currently require cmake 3.15)

**Comment by ras0219-msft** [comment]
Since there's a trivially correct fallback (don't use the PCH), I'm happy to start using it now under a guard. Thanks!

File: CMakeLists.txt [hunk 1]
```diff
@@ -365,20 +365,15 @@ target_link_libraries(vcpkglib
...
-
-    if(NOT CMAKE_GENERATOR MATCHES "Visual Studio .*")
-        set_property(SOURCE src/pch.cpp APPEND PROPERTY OBJECT_OUTPUTS "${CMAKE_CURRENT_BINARY_DIR}/pch.pch")
-        set_property(SOURCE ${_srcs} APPEND PROPERTY OBJECT_DEPENDS "${CMAKE_CURRENT_BINARY_DIR}/pch.pch")
-    endif()
-
-    set_source_files_properties(src/pch.cpp PROPERTIES COMPILE_FLAGS "/Ycpch.h")
-    target_sources(vcpkglib PRIVATE src/pch.cpp)
-    target_compile_options(vcpkglib PRIVATE /Yupch.h /FIpch.h /Zm200)
+    target_compile_options(vcpkglib PRIVATE /FIpch.h)
```
Comment (BillyONeal) [inline CMakeLists.txt:368]: I think the `/FI` should go away with `target_precompile_headers`.

File: CMakeLists.txt [hunk 2]
```diff
@@ -453,6 +448,16 @@ if (BUILD_TESTING)
         target_link_libraries(vcpkg-test PRIVATE log)
     endif()
 
+    if(MSVC)
+        target_compile_options(vcpkg-test PRIVATE /FIpch.h)
```
Comment (BillyONeal) [inline CMakeLists.txt:452]: Ditto

File: CMakeLists.txt [hunk 3]
```diff
@@ -453,6 +448,16 @@ if (BUILD_TESTING)
...
+    else()
+        target_compile_options(vcpkg-test PRIVATE -include "${CMAKE_CURRENT_SOURCE_DIR}/include/pch.h")
+    endif()
+
+    if(CMAKE_VERSION GREATER_EQUAL "3.16")
+        target_precompile_headers(vcpkg-test PRIVATE "include/pch.h")
```
Comment (BillyONeal) [inline CMakeLists.txt:458]:
```suggestion
        target_precompile_headers(vcpkg-test REUSE_FROM vcpkglib)
```

**Review by BillyONeal — COMMENTED** [review]
> Since there's a trivially correct fallback (don't use the PCH), I'm happy to start using it now under a guard. Thanks!

I don't think we have to worry about a guard: we ship binaries now and 3.15 is already newer than what comes in the box for lots of *nix.

**Comment by autoantwort** [comment]
I now get a lot of warnings like 
```
clang: warning: precompiled header '/Users/leanderSchulten/git_projekte/build-vcpkg-tool-No_QT-Debug/CMakeFiles/vcpkglib.dir/cmake_pch_arm64.hxx.pch' was ignored because '-include /Users/leanderSchulten/git_projekte/build-vcpkg-tool-No_QT-Debug/CMakeFiles/vcpkglib.dir/cmake_pch_arm64.hxx' is not first '-include'
```

**Comment by autoantwort** [comment]
I also have created #969 to fix the precompiled header warnings

**Pull request merged**